File: src/main/java/Model/RegistrationRequest.java
package Model;
import java.sql.Timestamp;
/*
 * CREATE TABLE RegistrationRequest (
 *    requestId INT AUTO_INCREMENT PRIMARY KEY,
 *    id_user VARCHAR(50),                             -- ID del usuario solicitante
 *    request_type ENUM('Profesor', 'Estudiante') NOT NULL,  -- Tipo de solicitud
 *    status ENUM('Pendiente', 'Aceptada', 'Denegada') DEFAULT 'Pendiente', -- Estado de la solicitud
 *    requested_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP,  -- Fecha de la solicitud
 *    FOREIGN KEY (id_user) REFERENCES User(id_user) ON DELETE CASCADE
 * );
 * 
 * @author kathe
 */
    enum RequestType {
       PROFESOR, ESTUDIANTE
    }

    enum RequestStatus {
       PENDIENTE, ACEPTADA, DENEGADA
    }

public class RegistrationRequest {
    private int requestId;
    private User id_user;
    private RequestType request_type; 
   // private String request_type; 
   // private String status; 
    private RequestStatus status;
    private Timestamp requested_at; 

    public RegistrationRequest(int requestId, User id_user, RequestType request_type, RequestStatus status, Timestamp requested_at) {
        this.requestId = requestId;
        this.id_user = id_user;
        this.request_type = request_type;
        this.status = status;
        this.requested_at = requested_at;
    }

    public User getId_user() {
        return id_user;
    }

    public void setId_user(User id_user) {
        this.id_user = id_user;
    }

    public RequestType getRequest_type() {
        return request_type;
    }

    public void setRequest_type(RequestType request_type) {
        this.request_type = request_type;
    }

    public RequestStatus getStatus() {
        return status;
    }

    public void setStatus(RequestStatus status) {
        this.status = status;
    }

    public Timestamp getRequested_at() {
        return requested_at;
    }

    public void setRequested_at(Timestamp requested_at) {
        this.requested_at = requested_at;
    }


    public int getRequestId() {
        return requestId;
    }

    public void setRequestId(int requestId) {
        this.requestId = requestId;
    }

    public User getIdUser() {
        return id_user;
    }

    public void setIdUser(User id_user) {
        this.id_user = id_user;
    }

    public Timestamp getRequestedAt() {
        return requested_at;
    }

    public void setRequestedAt(Timestamp requested_at) {
        this.requested_at = requested_at;
    }

    @Override
    public String toString() {
        return "RegistrationRequestAPI{" +
                "requestId=" + requestId +
                ", id_user=" + id_user +
                ", request_type='" + request_type + '\'' +
                ", status='" + status + '\'' +
                ", requested_at=" + requested_at +
                '}';
    }
}

File: flask-mvc-starter-kit/app/models/Registration_Request.py
# models/Registration_Request.py
from flask_sqlalchemy import SQLAlchemy
from app import db

class RegistrationRequest(db.Model):
    __tablename__ = 'RegistrationRequest'
    
    requestId = db.Column(db.Integer, primary_key=True, autoincrement=True)
    id_user = db.Column(db.String(50), db.ForeignKey('User.id_user', ondelete="CASCADE"), nullable=False)
    request_type = db.Column(db.Enum('Profesor', 'Estudiante'), nullable=False)
    status = db.Column(db.Enum('Pendiente', 'Aceptada', 'Denegada'), default='Pendiente')
    requested_at = db.Column(db.DateTime, default=db.func.current_timestamp())

    # Relaciones 
    user = db.relationship("User", backref=db.backref("registration_requests", lazy=True))

    def to_dict(self):
        return {
            "requestId": self.requestId,
            "id_user": self.id_user,
            "request_type": self.request_type,
            "status": self.status,
            "requested_at": self.requested_at,
            "user": self.user.to_dict() if self.user else None
        }